### ISSUE TITLE
Use one icon for adding location

### ISSUE THREAD

**Issue by GrimPixel** [breezy-weather/breezy-weather#519]
### Describe your idea

Currently, there are two icons for adding a location: a “magnifier” at the upper-right corner and a “positioning” icon at the other. The “magnifier” gives a sense of searching from added entries.
How about this: a “plus” icon; After being tapped on, a prompt with two fields to be defined shows up: location and service; The location can be defined with positioning.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] My idea is not related to a question covered in the [Frequently Asked Questions](https://github.com/breezy-weather/breezy-weather/blob/main/HELP.md) document, or a feature described in the [Homepage](https://github.com/breezy-weather/breezy-weather/blob/main/docs/HOMEPAGE.md) document.
- [X] I have updated the app to **[version 4.6.7 or later](https://github.com/breezy-weather/breezy-weather/releases/latest)**.
- [X] I understand that this app is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY, so I may not get a positive reply to my request, if any at all.

**Comment by papjul** [comment]
> a prompt with two fields to be defined shows up: location and service

I don't understand that part, how would that look like? Quick hand drawn mockup maybe?

**Comment by GrimPixel** [comment]
![ui](https://github.com/breezy-weather/breezy-weather/assets/39795763/21b8c9fa-32c1-4435-a1ae-592d112f6599)

**Comment by papjul** [comment]
Where are location results?

What is Service? If it is Source, source must be selected before location results as results are provided by the source.

**Comment by GrimPixel** [comment]
I think the search will be triggered like the current “magnifier” icon after tapping on that rectangle.
Yes, the “service” is the source. Currently, the source is not necessarily selected before the location, when the user taps on the “positioning” icon.

**Comment by papjul** [comment]
> I think the search will be triggered like the current “magnifier” icon after tapping on that rectangle.

I find it less intuitive than it currently is and it isn't following Material Design guidelines for search.

> Yes, the “service” is the source. Currently, the source is not necessarily selected before the location, when the user has the option to tap on the “positioning” icon.

Yes, but we are talking about location search. It is necessary because it depends on the source to provide location results for the query input by the user.

**Comment by GrimPixel** [comment]
So, it may look like this: after tapping on the “add” button, two prompts show up one after another: first select the source, then select the location (either by manual input or by positioning). This is based on the relation between “location” and “source”.

**Comment by papjul** [comment]
But you want to keep the search screen as it is, right? So you just add more screens and steps in the process. For current location, we went from 2 taps to 6 taps. For search, we went from 3 taps to 7 taps.

Additionally, being prompted to select a source first is not very "basic user"-friendly, I like how it is pre-selected for them.

Let's start over to the need before making mockups, what do you need exactly? What is causing you troubles exactly in the location list that you want to change it?

**Comment by GrimPixel** [comment]
The reason is room for optimisation.

Let's count every step we need currently, from start to end.

For adding manually:
1. [upper-right] “three locators”
2. [upper-right] “magnifier”
3. [lower-right] “sliders”
4. [center] option for source
5. [upper-center] search bar
6. [lower] keyboard
7. [upper] option for location

For positioning:
1. [upper-right] “three locators”
2. [lower-right] “positioning”
3. [center] option for source


After reconsideration, I think it could be like:

For adding manually:
1. [upper-right] “three locators”
2. [upper-right] “plus sign”
3. [center] option for source
4. [lower] keyboard
5. [lower] option for location
Sthep 5 means that the list is showing the most relevant results at the bottom and the focus is at the bottom.

For positioning:
1. [upper-right] “three locators”
2. [upper-right] “plus sign”
3. [center] option for source
4. [center] “positioning”
Or
1. [upper-right] “three locators”
2. [upper-right] “positioning”
3. [center] option for source

**Comment by papjul** [comment]
> The reason is room for optimisation.

Optimisation of what?

> Let's count every step we need currently, from start to end.

## Currently

Manual location
1. Search icon
(Optionally change source here)
2. Tap search bar
3. Tap the matching location result

Current location
1. FAB button
2. Select source

## Your suggestion

Manual location

1. Plus icon
2. Tap Source
3. Select source
4. Tap Next to go to 2nd screen
5. Tap Location
6. Tap search bar
7. Tap the matching location result
8. Tap Check button to add

Current location

1. Plus icon
2. Tap Source
3. Select source
4. Tap Next to go to 2nd screen
5. Tap Current location icon
6. Tap Check button to add

So it's not optimized at all.

**Comment by GrimPixel** [comment]
>Manual location

>1. Search icon
>(Optionally change source here)
>2. Tap search bar
>3. Tap the matching location result

OK. Let me show you:

<img src="https://github.com/breezy-weather/breezy-weather/assets/39795763/1b140f9b-2247-4ed7-b5be-b9c0ed9d46fa" width="200" />
<img src="https://github.com/breezy-weather/breezy-weather/assets/39795763/d6cbfe05-eee9-49da-9dde-bac9d3f2338c" width="200" />
<img src="https://github.com/breezy-weather/breezy-weather/assets/39795763/961f72a8-7dbc-4d80-bc32-dc34af8c08c1" width="200" />
<img src="https://github.com/breezy-weather/breezy-weather/assets/39795763/0b76c944-30ea-42d5-9532-3d53f476161f" width="200" />
<img src="https://github.com/breezy-weather/breezy-weather/assets/39795763/d310e4f7-e978-4882-a9e5-027ed324c6d3" width="200" />
<img src="https://github.com/breezy-weather/breezy-weather/assets/39795763/21c7900a-ba2e-4d91-9978-8f80baad4d76" width="200" />
<img src="https://github.com/breezy-weather/breezy-weather/assets/39795763/ad7140ed-341d-40bf-9cb0-6949e581fcd8" width="200" />
<img src="https://github.com/breezy-weather/breezy-weather/assets/39795763/52f80345-f1da-4eab-b4fa-b216ffff8be3" width="200" />
<img src="https://github.com/breezy-weather/breezy-weather/assets/39795763/1e5233b0-f7b1-47e8-8565-908737cd37d9" width="200" />

**Comment by papjul** [comment]
Your screenshots are exactly what I described, with the addition of the optional steps. So 3 taps, with 2 optional taps. (Starting from the location list)

The steps I put are based on your initial mockup, followed by your reconsideration as a two step screen you suggested after I explained we needed source first. Quoting your reconsideration after I explained that:

> after tapping on the “add” button, two prompts show up one after another: first select the source, then select the location

You wrote a step-by-step that misses some steps from this reconsideration, so I put all taps in my reply, showing that your suggestions make the navigation more painful by having more taps. So I didn't misinterpreted, and I'm not making this app for you or your friends, it's a FOSS app, it's for myself and for whoever finds it useful. Quoting LGPL 3.0 licence you agreed with when installing:

> Breezy Weather is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

I'm not paid to do this, I spend a lot of time doing free support (which is a time I don't spend developing or doing the things I like) and implementing community requests, so you have to be cooperative to not make me lose time for nothing: if anything must be improved, it must be for everyone, so ignoring my question or making evasive answers regarding the need is not respectful of the dev that is maintaining this app out of consideration for the community.

Tl;dr: What I'm asking is to stop pushing solutions and explain the need in details first, which is essential to understand the issue and find a common ground that will please most people ;)

**Comment by GrimPixel** [comment]
OK, I am here to help with ideas. I will do it in detail this time. My mistake was not distinguishing required and optional taps., I am here to help with ideas. I will do it in detail this time. My mistake was not distinguishing required and optional taps.

Here are steps, from the beginning to the end:
# 1.
<img src="https://github.com/breezy-weather/breezy-weather/assets/39795763/22649a32-564e-4570-871b-e323d7077d19" width="300" />

# 2.
<img src="https://github.com/breezy-weather/breezy-weather/assets/39795763/a07cf919-679f-469b-ba45-3e7a875cf75f" width="300" />

# 3.
<img src="https://github.com/breezy-weather/breezy-weather/assets/39795763/3911b6a8-a554-4fa3-bdd3-20b0e49b676a" width="300" />

# 4-1
<img src="https://github.com/breezy-weather/breezy-weather/assets/39795763/2dca261a-2663-47a0-b810-bb9873616dfb" width="300" />

# 4-2
<img src="https://github.com/breezy-weather/breezy-weather/assets/39795763/d3cc2575-2234-474d-a753-2c22db867db8" width="300" />
<img src="https://github.com/breezy-weather/breezy-weather/assets/39795763/8798967b-24e0-4094-a697-86bc27bf316f" width="300" />

# 4-3
<img src="https://github.com/breezy-weather/breezy-weather/assets/39795763/b14562b7-8edf-44d5-a1ca-11ecee0b254a" width="300" />

# 5
<img src="https://github.com/breezy-weather/breezy-weather/assets/39795763/6db6c451-910b-498b-8acf-af6214f2482b" width="300" />

**Comment by papjul** [comment]
Thank you for the mockups, however for the 4th time now, you don't answer about the need.

1. I don't know how you go from 4-1 to 4-2.
2. You put source and location on the same screen. However, what happens if someone changes the source after choosing a location? That case is not possible.
3. I think you miss a button to submit the form.
4. If you want the keyboard to pop automatically, I can do that, that would reduce the current process to 2 required taps in both manual and current location.

**Comment by GrimPixel** [comment]
Thank you for your patience.
1. It's from 3 to 4-1 or 4-2 or 4-3.
2. I thought there could be a saved list of cities from those sources. If it's not the case, then it doesn't need to be instant.
3. Then clear the location input.
4. There is no “submit” button, because once the location is selected, either way, it jumps directly to 5.
5. Great to hear that.
